agent/src/beerocks/monitor/monitor_stats.cpp

```diff
@@ -159,13 +183,31 @@ void monitor_stats::process()
 
             auto &sta_stats = sta_node->get_stats();
 
-            if (!response->alloc_sta_stats()) {
-                LOG(ERROR) << "tx_buffer overflow!";
-                return;
+            if (elements_to_allocate == sta_count) {
+                message_com::send_cmdu(slave_socket, cmdu_tx);
+                response = message_com::create_vs_message<
+                    beerocks_message::cACTION_MONITOR_HOSTAP_STATS_MEASUREMENT_RESPONSE>(cmdu_tx);
+                if (!response) {
+                    LOG(ERROR) << "Failed building "
+                                  "cACTION_MONITOR_HOSTAP_STATS_MEASUREMENT_RESPONSE message!";
+                    return;
+                }
+
+                total_sta_count -= sta_count;
+                elements_to_allocate = cmdu_tx.mtu_size_restriction_overflow_check(
+                    sizeof(beerocks_message::sStaStatsParams));
+
+                elements_to_allocate = std::min(total_sta_count, elements_to_allocate);
+
+                if (!response->alloc_sta_stats(elements_to_allocate)) {
+                    LOG(ERROR) << "tx_buffer overflow!";
+                    return;
+                }
```

Comment:
All this code is a duplication.
Consider wrapping it with a local lambda function and just calling it twice.

Reply:
ok